Stripe/STPAPIClient.m

```diff
@@ -260,9 +260,42 @@ - (void)createTokenWithPersonalIDNumber:(NSString *)pii completion:(__nullable S
 
 @implementation STPAPIClient (Upload)
 
+- (NSData *)dataForUploadedImage:(UIImage *)image
```

Comment:
@bdorfman-stripe - any luck with doing this heuristically? (Based on max JPEG file sizes for certain resolutions?) Also, to test this, you could pass through to an internal `dataForUploadedImage:maxFileSize:` method, and then test that with a couple small test images.